corehq/pillows/case_search.py

```diff
@@ -94,12 +89,9 @@ def _add_smart_types(dynamic_properties, domain, case_type):
     # Properties are stored in a dict like {"key": "dob", "value": "1900-01-01"}
     # `value` is a multi-field property that duck types numeric and date values
     # We can't do that for properties like geo_points in ES v2, as `ignore_malformed` is broken
-    if USH_CASE_CLAIM_UPDATES.enabled(domain):
-        gps_props = get_gps_properties(domain, case_type)
-        _add_gps_smart_types(dynamic_properties, gps_props)
-    if GEOSPATIAL.enabled(domain):
-        gps_props = [get_geo_case_property(domain)]
-        _add_gps_smart_types(dynamic_properties, gps_props)
+    gps_props = get_gps_properties(domain, case_type)
+    gps_props.add(get_geo_case_property(domain))
+    _add_gps_smart_types(dynamic_properties, gps_props)
```

Comment:
you could skip calling the function if there are no gps_props?

Reply:
That's a good idea, this would save on some unnecessary loops. Addressed in [d957c7d](https://github.com/dimagi/commcare-hq/pull/35001/commits/d957c7d751bc008141beeb58480872e26ea9c5b9).

Reply:
FYI the above commit has been removed as part of a rebase. See [this](https://github.com/dimagi/commcare-hq/pull/35001#discussion_r1726678049) comment for more info.

Reply:
Okay, I was looking for the classic  mock principle -  'PATCH where it is used and not where it is defined'.
As the decorator does not has the function used  in the file patched [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/util/test_utils.py#L181), simply adding 'corehq.pillows.case_search.domain_has_privilege' should resolve the issue.

Reply:
Ah, good catch @ajeety4! Having tested this locally it does indeed resolve the issue and the tests are passing correctly now with using the `privilege_enabled` decorator.

Reply:
Glad it worked. Just realised I had replied in the wrong thread :p